tests/Unit/Infrastructure/Auth/RoleAccessTest.php

```diff
@@ -55,23 +55,60 @@ public function testReturnsNothingIfCheckSucceededAndUserHasAdminPermission()
         $auth = Mockery::mock(Authentication::class);
         $auth->shouldReceive('check')->andReturn(true);
         $auth->shouldReceive('user')->andReturn($user);
-        $this->swap(Authentication::class, $auth);
 
         //The middleware doesn't do anything if the user is an admin, so it returns null (void)
-        $this->assertNull(RoleAccess::userHasAccess($this->app, 'admin'));
+        $this->assertNull(RoleAccess::userHasAccess($auth, 'admin'));
     }
 
     public function testReviewerCantGetToAdminPages()
     {
-        $this->asReviewer();
-        $this->assertInstanceOf(RedirectResponse::class, RoleAccess::userHasAccess($this->app, 'admin'));
-        $this->assertNull(RoleAccess::userHasAccess($this->app, 'reviewer'));
+        $user = Mockery::mock(UserInterface::class);
+        $user->shouldReceive('hasAccess')->with('reviewer')->andReturn(true);
+        $user->shouldReceive('hasAccess')->with('admin')->andReturn(false);
+
+        $auth = Mockery::mock(Authentication::class);
+        $auth->shouldReceive('check')->andReturn(true);
+        $auth->shouldReceive('user')->andReturn($user);
+
+        $this->assertInstanceOf(RedirectResponse::class, RoleAccess::userHasAccess($auth, 'admin'));
+    }
+
+    public function testReviewerCanGetToReviewerPage()
+    {
+        $user = Mockery::mock(UserInterface::class);
+        $user->shouldReceive('hasAccess')->with('reviewer')->andReturn(true);
+        $user->shouldReceive('hasAccess')->with('admin')->andReturn(false);
+
+        $auth = Mockery::mock(Authentication::class);
+        $auth->shouldReceive('check')->andReturn(true);
+        $auth->shouldReceive('user')->andReturn($user);
+
+        $this->assertNull(RoleAccess::userHasAccess($auth, 'reviewer'));
     }
 
     public function testAdminCantGetToReviewerPage()
     {
-        $this->asAdmin();
-        $this->assertInstanceOf(RedirectResponse::class, RoleAccess::userHasAccess($this->app, 'reviewer'));
-        $this->assertNull(RoleAccess::userHasAccess($this->app, 'admin'));
+        $user = Mockery::mock(UserInterface::class);
+        $user->shouldReceive('hasAccess')->with('admin')->andReturn(true);
+        $user->shouldReceive('hasAccess')->with('reviewer')->andReturn(false);
+
+        $auth = Mockery::mock(Authentication::class);
+        $auth->shouldReceive('check')->andReturn(true);
+        $auth->shouldReceive('user')->andReturn($user);
+
+        $this->assertInstanceOf(RedirectResponse::class, RoleAccess::userHasAccess($auth, 'reviewer'));
+    }
+
+    public function testAdminCanGetToAdminPage()
+    {
+        $user = Mockery::mock(UserInterface::class);
+        $user->shouldReceive('hasAccess')->with('admin')->andReturn(true);
+        $user->shouldReceive('hasAccess')->with('reviewer')->andReturn(false);
```

Comment:
Can we remove this expectation? The method `User::hasAccess()` is only invoked with `'admin'` in the context of this test.